provisioning/baremetal_secrets.go

```diff
@@ -168,3 +152,72 @@ password = %s
 	_, err = client.Secrets(targetNamespace).Create(context.Background(), secret, metav1.CreateOptions{})
 	return err
 }
+
+// CreateTlsSecret creates a Secret for the Ironic and Inspector TLS
+func CreateTlsSecret(client coreclientv1.SecretsGetter, targetNamespace string, baremetalConfig *metal3iov1alpha1.Provisioning, scheme *runtime.Scheme) error {
+	existing, err := client.Secrets(targetNamespace).Get(context.Background(), tlsSecretName, metav1.GetOptions{})
+	if err == nil {
+		changed := false
+
+		if len(existing.ObjectMeta.OwnerReferences) == 0 {
+			err = controllerutil.SetControllerReference(baremetalConfig, existing, scheme)
+			if err != nil {
+				return err
+			}
+			changed = true
+		}
+
+		existingCert := existing.StringData[tlsCertificateKey]
+		expired, err := isTlsCertificateExpired(existingCert)
+		if err != nil {
+			return err
+		}
+
+		if expired {
+			cert, err := generateTlsCertificate(baremetalConfig.Spec.ProvisioningIP)
+			if err != nil {
+				return err
+			}
+			existing.StringData[tlsCertificateKey] = cert.certificate
+			existing.StringData[tlsPrivateKeyKey] = cert.privateKey
+			changed = true
+		}
+
+		if changed {
+			_, err = client.Secrets(targetNamespace).Update(context.Background(), existing, metav1.UpdateOptions{})
+			return err
+		} else {
+			return nil
+		}
+	}
+
+	if !apierrors.IsNotFound(err) {
```

Comment:
In go it is more common to exit early when there is an error. Moving this block up to line 159 would be consistent with that style advice.